lib/Sema/CSFix.cpp

```diff
@@ -858,6 +858,10 @@ CollectionElementContextualMismatch::create(ConstraintSystem &cs, Type srcType,
 bool DefaultGenericArgument::coalesceAndDiagnose(
     const Solution &solution, ArrayRef<ConstraintFix *> fixes,
     bool asNote) const {
+  
+  if (fixes.empty() && diagnoseForKeyPathRootRepresentatee(solution, asNote))
```

Comment:
Unfortunately this is too late to do this since constraint system at this point is going to be back to its original state (currently we still call `applySolution` but that's temporary).W we need to determine whether there is an equivalence established when we are about to bind generic parameter to a hole and record a different fix...

Reply:
> Unfortunately this is too late to do this since constraint system at this point is going to be back to its original state (currently we still call applySolution but that's temporary).

Yeah, that is why I computed the key path representatee on the fix constructor we have to do this before the solver scope cleans up the constraint system. 

> we need to determine whether there is an equivalence established when we are about to bind generic parameter to a hole and record a different fix...

Right! I'll make the changes :)